src/libYARP_dev/include/yarp/dev/IJoypadController.h

```diff
@@ -253,8 +253,11 @@ class YARP_dev_API yarp::dev::IJoypadEventDriven : yarp::os::RateThread,
 
 
     IJoypadEventDriven();
-
-    IJoypadEventDriven(int rate);
+#ifndef YARP_NO_DEPRECATED // Since YARP 3.0.0
+    YARP_DEPRECATED_MSG("Use IJoypadEventDriven(double)")
+    explicit IJoypadEventDriven(int rate);
+#endif
```

Comment:
Are you sure that `YARP_NO_DEPRECATED` will be defined in all the compilation units that include `IJoypadController.h`? If this is not the case, the ABI of this interface changes if `YARP_NO_DEPRECATED` is used, and this could create runtime problems. Even if this define is properly exported in the YARP cmake imported targets, this would break all the downstream consumers of YARP that do not use CMake.

Reply:
`YARP_NO_DEPRECATED` is defined in `YARPConfig.cmake`

https://github.com/robotology/yarp/blob/80debe512994fbc630178597c65a8fc73c9b3040/cmake/template/YARPConfig.cmake.in#L27:L32

Therefore, if using CMake, unless you are doing something nasty, there is no way to change its value.

Anyway, I realize that a better is probably to move it in the api.h file, to ensure that it is always defined, even if YARP is linked manually...

Reply:
Ok!